src/corecel/data/Collection.hh

```diff
@@ -294,7 +294,7 @@ class Collection
     //! Direct accesors to underlying data
     CELER_FORCEINLINE_FUNCTION size_type size() const
     {
-        return this->storage().size();
+        return static_cast<size_type>(this->storage().size());
```

Comment:
why is the return type of `.size()` not a `size_type`?  Can it cause (actual) problems?

Reply:
The underlying data container can be a `Span` or `std::vector` (both `std::size_t`) or `DeviceVector` (`celeritas::size_type`), but more importantly, the `size_type` of the Collection is the `size_type` of the `OpaqueId` accessor, which could be much smaller. Thinking about it now, that may not be the best policy (we should prefer not to be allocating "small" arrays), but I *do* have logic in the collection builder and device copier to ensure that we don't exceed the size of the collection.

Reply:
Fair enough.